### PR TITLE
[Feature] using new Parameters attribute instead of Pass state.

### PR DESCRIPTION
**Pull request by grahamjenson**
This PR uses the new Parameters attributes to implement `TaskFn` rather than using a prior Pass State. This reduces the number of states, cleans up visualisations, reduces cost to about nearly 50% in some cases.

This PR also:
1.  changes `ResourceFunction` to `TaskHandler` as it is more descriptive
2. Improves Readme
3. changes how some testing is done

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -199,10 +177,12 @@ step deploy                           \
 
 ### Development State
 
-Step is still very Beta and its API might change quickly.
```
Comment (samnrubin) [inline README.md]: 🤣

**Comment by samnrubin** [comment]
PR looks good to me but why are the [exec instructions being removed?](https://github.com/coinbase/step/pull/33/files#diff-04c6e90faac2675aa89e2176d2eec7d8L117) I might be confused but I can't find any change explicitly removing that functionality

File: README.md (hunk 1 above)
Comment (grahamjenson) [reply README.md]: now it is only Beta, not very Beta :)

**Comment by grahamjenson** [comment]
> PR looks good to me but why are the [exec instructions being removed?](https://github.com/coinbase/step/pull/33/files#diff-04c6e90faac2675aa89e2176d2eec7d8L117) I might be confused but I can't find any change explicitly removing that functionality

I implemented `exec` but found it to not be that useful other than for really simple use-cases. Leaving the functionality in but not dedicating so much README space to it was the idea.

**Pull request merged**